From 778869fa139ab72ed557e7455e4f1126684f2625 Mon Sep 17 00:00:00 2001
From: Andriy Gapon <avg@FreeBSD.org>
Date: Thu, 18 Feb 2021 23:53:05 +0200
Subject: [PATCH 10/38] Fix report_mount_progress never calling
 set_progress_header

That happens because of an off-by-one mistake.
share_mount_one_cb() calls report_mount_progress(current=sm_done) after
having incremented sm_done by one.  Then report_mount_progress()
increments the parameter again.  It appears that that logic became
obsolete after commit a10d50f999511, parallel zfs mount.

On FreeBSD I observe that zfs mount -a -v prints, for example,
    (null): (201/248)
That happens because set_progress_header() is never called.

With this change the output becomes correct:
    Mounting ZFS filesystems: (209/248)

Reviewed-by: Brian Behlendorf <behlendorf1@llnl.gov>
Signed-off-by: Andriy Gapon <avg@FreeBSD.org>
Closes #11607
---
 cmd/zfs/zfs_main.c | 3 ---
 1 file changed, 3 deletions(-)

Index: zfs/cmd/zfs/zfs_main.c
===================================================================
--- zfs.orig/cmd/zfs/zfs_main.c
+++ zfs/cmd/zfs/zfs_main.c
@@ -6790,9 +6790,6 @@ report_mount_progress(int current, int t
 	time_t now = time(NULL);
 	char info[32];
 
-	/* report 1..n instead of 0..n-1 */
-	++current;
-
 	/* display header if we're here for the first time */
 	if (current == 1) {
 		set_progress_header(gettext("Mounting ZFS filesystems"));
